Source/Projections/Builder/ClassProjectionMethod.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+using Dolittle.SDK.Async;
+using Dolittle.SDK.Events;
+
+namespace Dolittle.SDK.Projections.Builder
+{
+    /// <summary>
+    /// An implementation of <see cref="IProjectionMethod{TProjection}" /> that invokes a method on a projection instance for an event of a specific type.
+    /// </summary>
+    /// <typeparam name="TProjection">The <see cref="Type" /> of the projection.</typeparam>
+    public class ClassProjectionMethod<TProjection> : IProjectionMethod<TProjection>
+        where TProjection : class, new()
+    {
+        readonly TaskResultProjectionMethodSignature<TProjection> _method;
+        readonly EventType _eventType;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ClassProjectionMethod{TEventHandler}"/> class.
+        /// </summary>
+        /// <param name="method">The <see cref="TaskProjectionMethodSignature{TProjection}"/> method to invoke.</param>
+        /// <param name="eventType">The <see cref="EventType" /> of the event.</param>
+        /// <param name="keySelector">The <see cref="Projections.KeySelector"/> for selecting the key.</param>
+        public ClassProjectionMethod(TaskProjectionMethodSignature<TProjection> method, EventType eventType, KeySelector keySelector)
+            : this(
+                (TProjection instanceAndReadModel, object @event, ProjectionContext context)
+                    => method(instanceAndReadModel, @event, context).ContinueWith(_ => ProjectionResultType.Replace, TaskScheduler.Default),
```

Comment:
Since I'm (still) not an TPL expert, does this line behave the same way as an async function with await in it? If we're not sure, maybe best to just stick with the pattern we know (based on the little trouble we've had recently).

Reply:
Yeah this should do the same thing, but as you say async/await is more familiar